src/pages/serviceList.tsx

```diff
@@ -24,26 +27,34 @@ interface ServiceListProps {
     buttonText: string;
     errors: ErrorInfo[];
     multiOperator: boolean;
+    dataSourceAttribute: TxcSourceAttribute;
     csrfToken: string;
 }
 
-const ServiceList = ({ serviceList, buttonText, csrfToken, errors, multiOperator }: ServiceListProps): ReactElement => (
+const ServiceList = ({
+    serviceList,
+    buttonText,
+    csrfToken,
+    errors,
+    multiOperator,
+    dataSourceAttribute,
+}: ServiceListProps): ReactElement => (
     <FullColumnLayout title={pageTitle} description={pageDescription}>
+        <SwitchDataSource dataSourceAttribute={dataSourceAttribute} pageUrl="/serviceList" csrfToken={csrfToken} />
         <CsrfForm action="/api/serviceList" method="post" csrfToken={csrfToken}>
             <>
                 <ErrorSummary errors={errors} />
                 <div className={`govuk-form-group ${errors.length > 0 ? 'govuk-form-group--error' : ''}`}>
+                    <legend className="govuk-fieldset__legend govuk-fieldset__legend--s">
+                        <h1 className="govuk-heading-l" id="service-list-page-heading">
+                            Which {multiOperator ? 'of your ' : ''}services is the ticket valid for?
+                        </h1>
+                    </legend>
+
+                    <span className="govuk-heading-s">
+                        Select all {multiOperator ? 'of your ' : ''}services that apply
+                    </span>
                     <fieldset className="govuk-fieldset">
-                        <legend className="govuk-fieldset__legend govuk-fieldset__legend--s">
-                            <h1 className="govuk-heading-l" id="service-list-page-heading">
-                                Which {multiOperator ? 'of your ' : ''}services is the ticket valid for?
-                            </h1>
-                        </legend>
-
-                        <span className="govuk-heading-s">
-                            Select all {multiOperator ? 'of your ' : ''}services that apply
-                        </span>
-
                         <input
                             type="submit"
                             name="selectAll"
```

Comment:
It won't let me comment on it as it's not expanded by default (didn't realise that was a thing), but should line 66 be changed to:
```
This data is taken from the{' '}
<b>
    {dataSourceAttribute.source === 'tnds'
        ? 'Traveline National Dataset (TNDS)'
        : 'Bus Open Data Service (BODS)'}
</b>
. If the service you are looking for is not listed, contact the BODS help desk for advice{' '}
<a href="/contact">here</a>.
```